content/blog/2018-03-15-update-on-async-rendering.md

```diff
@@ -0,0 +1,168 @@
+
+We'll look at examples of how both of these lifecycles can be used below.
+
+> Note
+>
+> For brevity, the examples below are written using the experimental class properties transform, but the same migration strategies apply without it.
+
+---
+
+### Initializing state
+
+This example shows a component with `setState` calls inside of `componentWillMount`:
+`embed:update-on-async-rendering/initializing-state-before.js`
+
+The simplest refactor for this type of component is to move state initialization to the constructor or to a property initializer, like so:
+`embed:update-on-async-rendering/initializing-state-after.js`
+
+### Fetching external data
+
+Here is an example of a component that uses `componentWillMount` to fetch external data:
+`embed:update-on-async-rendering/fetching-external-data-before.js`
+
+The above code is problematic for both server rendering (where the external data won't be used) and the upcoming async rendering mode (where the request might be initiated multiple times).
+
+The recommended upgrade path for most use cases is to move data-fetching into `componentDidMount`:
+`embed:update-on-async-rendering/fetching-external-data-after.js`
+
+There is a common misconception that fetching in `componentWillMount` lets you avoid the first empty rendering state. In practice this was never true because React has always executed `render` immediately after `componentWillMount`. If the data is not available by the time `componentWillMount` fires, the first `render` will still show a loading state regardless of where you initiate the fetch. This is why moving the fetch to `componentDidMount` has no perceptible effect in the vast majority of cases.
+
+> Note
+>
+> Some advanced use-cases (e.g. libraries like Relay) may want to experiment with eagerly prefetching async data. An example of how this can be done is available [here](https://gist.github.com/bvaughn/89700e525ff423a75ffb63b1b1e30a8f).
+
+### Adding event listeners (or subscriptions)
+
+Here is an example of a component that subscribes to an external event dispatcher when mounting:
+`embed:update-on-async-rendering/adding-event-listeners-before.js`
+
+Unfortunately, this can cause memory leaks for server rendering (where `componentWillUnmount` will never be called) and async rendering (where rendering might be interrupted before it completes, causing `componentWillUnmount` not to be called).
+
+People often assume that `componentWillMount` and `componentWillUnmount` are always paired, but that is not guaranteed. Only once `componentDidMount` has been called does React guarantee that `componentWillUnmount` will later be called for clean up.
+
+For this reason, the recommended way to add listeners/subscriptions is to use the `componentDidMount` lifecycle:
+`embed:update-on-async-rendering/adding-event-listeners-after.js`
+
+> Note
+>
```

Comment:
Instead of these notes, should we just plug `create-subscription` and then show an equivalent example with it?

Reply:
Yeah. I think that's a reasonable change.

Reply:
I took a stab at this. Let me know what you think?

It's a little awkward since `create-subscription` won't be released to NPM (and thus won't have a README) until we publish 16.3- so I had to link to the GitHub folder.